qt/python/mantidqt/mantidqt/widgets/codeeditor/test/test_multifileinterpreter.py

```diff
@@ -49,6 +51,24 @@ def test_open_file_in_new_tab_no_import_added(self):
                 widget.open_file_in_new_tab(test_string)
         self.assertNotIn(MANTID_API_IMPORT, widget.current_editor().editor.text())
 
+    def test_save_as_closes_the_original_tab(self):
+        test_string = "Test file\n"
+        widget = MultiPythonFileInterpreter()
+        mock_open_func = mock.mock_open(read_data=test_string)
+        with mock.patch(widget.__module__ + ".open", mock_open_func, create=True):
```

Comment:
Is it possible to do this without the nesting, e.g. https://stackoverflow.com/questions/43130969/are-multiple-with-statements-on-one-line-equivalent-to-nested-with-statement